apps/admin_app/lib/admin_app_web/helpers.ex

```diff
@@ -122,30 +158,35 @@ defmodule AdminAppWeb.Helpers do
     end
   end
 
-  def order_xlsx_exporter(user) do
-    orders = Repo.all(Order) |> Repo.preload([:line_items])
+  def xlsx_exporter(user, type) do
+    data_list =
+      case type do
+        "order" ->
+          Repo.all(Order) |> Repo.preload([:line_items])
+
+        "product" ->
+          Repo.all(Product)
+          |> Repo.preload([:variants, :taxon, :shipping_category, :options, :brand, :theme])
+      end
 
-    xlsx_generator(orders)
-    |> Elixlsx.write_to("/tmp/order.xlsx")
+    binary_data =
+      xlsx_generator(data_list, type)
+      |> Elixlsx.write_to("/tmp/#{type}s.xlsx")
 
-    attachment = "/tmp/order.xlsx"
+    attachment = "/tmp/#{type}s.xlsx"
 
-    DataExportMail.data_export_mail(attachment, user, "xlsx")
+    DataExportMail.data_export_mail(attachment, user, "xlsx", type)
   end
 
-  def xlsx_generator(orders) do
-    columns =
-      ~w(id number line_items_count order_total billing_address shipping_address inserted_at updated_at user_id state)
+  def xlsx_generator(data_list, type) do
+    columns = get_columns(type) |> Enum.map(&Atom.to_string(&1))
 
-    orders = orders |> Enum.map(&parse_line(&1))
-    rows = orders |> Enum.map(&row(&1))
-    %Workbook{sheets: [%Sheet{name: "Orders", rows: [columns] ++ rows}]}
+    data_list = data_list |> Enum.map(&parse_line(&1))
+    rows = data_list |> Enum.map(&row(&1, columns))
```

Comment:
IMO this can written simply as:-

```
rows = data_list 
|> Enum.map(&parse_line(&1))
|> Enum.map(&row(&1, columns))
```

what do you think? @jyotigautam

Reply:
Refactoring it this way.